webapp/channels/src/components/admin_console/custom_plugin_settings/enable_plugin_setting.ts

```diff
@@ -1,29 +1,19 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import type {PluginRedux} from '@mattermost/types/plugins';
+import type {PluginRedux, PluginSetting} from '@mattermost/types/plugins';
 
-import {Constants} from 'utils/constants';
 import {t} from 'utils/i18n';
 
+import type {AdminDefinitionSetting} from '../admin_definition';
 import SchemaAdminSettings from '../schema_admin_settings';
 
-export type EnabledPluginSetting = {
-    type: string;
-    key: string;
-    label: string;
-    label_default: string;
-    help_text: string;
-    help_text_default: string;
-    isDisabled?: () => boolean;
-}
-
-export default function getEnablePluginSetting(plugin: PluginRedux): EnabledPluginSetting {
+export default function getEnablePluginSetting(plugin: PluginRedux): Partial<AdminDefinitionSetting & PluginSetting> {
     const escapedPluginId = SchemaAdminSettings.escapePathPart(plugin.id);
     const pluginEnabledConfigKey = 'PluginSettings.PluginStates.' + escapedPluginId + '.Enable';
 
     return {
-        type: Constants.SettingsTypes.TYPE_BOOL,
```

Comment:
2/5 I'm not sure if I like getting rid of the constants because they're very generic string literals that would be hard to search through the code for. Would it be possible to keep using the constants for these if we change `SettingsTypes` to an enum or change it to use `as const`?

If that's not possible, then can we remove that constant now?

Reply:
I think this can be removed later, now is also used by other things like `schema_admin_settings.jsx` that is hard to migrate. Probably I'll migrate that after this gets in, because I don't want to make too many changes together.

Reply:
Yeah, that's probably fine. I'm not a huge fan of temporarily getting rid of the constants from these files, but at least it should be easier to switch those back to constants now that we have proper type definitions for that field